### ISSUE TITLE
Refactoring of DPU view in frontend

### ISSUE THREAD

**Issue by janvojt** [mff-uk/odcs#384]
When browsing code I noticed a terrible implementation of getting DPU instances for given DPU template. To find relevant instances, **all** DPU instances in application are iterated in a for loop. Furthermore, to get a list of Pipelines using these DPUs, all pipelines in the application are iterated over in a nested for loop. As a result, there are **3 nested for loops** iterating among pretty much all objects in the database. On top of it, there is some common logic in methods saveTableData() and deleteDPU() that should be merged into a single method. See [relevant TODO](https://github.com/mff-uk/intlib/blob/4c8792d3040520b600ae817ae2a07ca7f8dbb595/frontend/src/main/java/cz/cuni/xrg/intlib/frontend/gui/views/DPU.java#L1022).

I already added a method getPipelinesUsingDPU into PipelineFacade, which can be used here. If there are any more methods that are needed in commons-app, please create an issue and assign me.

This really needs to be refactored. Ideally, it should be done before issue #209. This test case includes 500+ pipelines. Let's say an average pipeline would have 5 DPU instances, and we end up with 2500 \* 500 \* 5 = **6.25M iterations!!**, which is unbearable with hibernate loading all these objects.

Tom please plan this issue and assign someone (Maria is the original author).

**Comment by tomas-knap** [comment]
Honzo, thanks for reporting this. Bohuslav, can you please check that? I am adding that to your list with high prio. If you need any method from Honzo, please ask him. If you think it is not doable till Monday, please discuss with Petr/Honza, if he can help you with that
